smartsim/settings/base.py

```diff
@@ -158,6 +163,27 @@ def add_exe_args(self, args):
                 raise TypeError("Executable arguments should be a list of str")
             self.exe_args.append(arg)
 
+    def set(self, arg, value=None):
```

Comment:
let's add a ``condition`` argument which will be used to make driver scripts more portable. 

example
```python
launcher = "slurm"
rs = exp.create_run_settings("echo", hello")
rs.set("exclusive", None, condition=bool(launcher=="slurm")
```


example
```python
import socket
hostname = socket.gethostname()
rs = exp.create_run_settings("echo", hello")
rs.set("exclusive", None, condition=bool(hostname=="theta")
```

I think the condition should default to True. 

thoughts on this @MattToast ?

Reply:
Love that idea! Good syntactic sugar that save the monotony of
```py
rs: RunSettings = ...
condition: bool = ...

if condition:
    rs.set("something")
```
added https://github.com/CrayLabs/SmartSim/pull/166/commits/a2d3a9414cced0fcc3a32b89229b54abf3fcbfdf

Reply:
I also decided to log a messages saying that the argument was not set because a condition was not met at the INFO level. Lmk if you think that the message should instead be at DEBUG or simply not there at all.